src/prog/NLOVEV.cpp

```diff
@@ -301,3 +192,122 @@ bool CLIOptions::good() const
   }
   return true;
 }
+
+CLIOptions::CLIOptions(const BSMPT::parser &argparser)
+{
+  argparser.check_required_parameters();
+  Model      = BSMPT::ModelID::getModel(argparser.get_value("model"));
+  InputFile  = argparser.get_value("input");
+  OutputFile = argparser.get_value("output");
+  FirstLine  = std::stoi(argparser.get_value("FirstLine"));
+  LastLine   = std::stoi(argparser.get_value("LastLine"));
+
+  try
+  {
+    UseGSL = argparser.get_value_lower_case("UseGSL") == "true";
+  }
+  catch (BSMPT::parserException &)
+  {
+  }
+
+  try
+  {
+    UseCMAES = argparser.get_value_lower_case("UseCMAES") == "true";
+  }
+  catch (BSMPT::parserException &)
+  {
```

Comment:
empty catch?!

Reply:
Yes, because this is an optional parameter and therefore it is possible for it not to exist. That's why I added a custom exception class to specialize catch